src/globus_sdk/login_flows/command_line_login_flow_manager.py

```diff
@@ -128,4 +137,16 @@ def prompt_for_code(self) -> str:
         :returns: The authorization code entered by the user.
         """
         code_prompt = "Enter the resulting Authorization Code here: "
-        return input(code_prompt).strip()
+        with self.handle_input_errors():
```

Comment:
Should this handler be up in `run_login_flow`, rather than here?

Here's the use-case that I'm thinking about:
```python
import contextlib

import click
from globus_sdk.experimental.login_flow_manager import CommandLineLoginFlowManager


class ClickLoginFlowManager(CommandLineLoginFlowManager):
    def print_authorize_url(self, authorize_url: str) -> None:
        click.echo(click.style("Login here for a code:", fg="yellow"))
        click.echo(authorize_url)

    def prompt_for_code(self) -> str:
        return click.prompt("Enter the code here:")

    @contextlib.contextmanager
    def handle_input_errors(self) -> t.Iterator[None]:
        try:
            yield
        except click.Abort as e
            raise CommandLineLoginFlowEOFError("User aborted when prompted for input.") from e
```

I don't have a clear-cut answer on this, but I think it bears consideration because it fundamentally asks the question:
What is the contract declared by `handle_input_errors`?

Reply:
In the original discussion, we landed on keeping `handle_input_errors` narrowly scoped to exactly that - handling errors that can happen with `input` - and calling it inside `prompt_for_code`. That was so that we could avoid trying to predict what kinds of errors might happen in `prompt_for_code` overrides, and instead let consumers handle errors that they need to handle as they come up. (And of course we would still provide `handle_input_errors` as a convenience method if anyone wants to use it.)

I also believe @derek-globus was specifically thinking about an SMTP-based implementation that would be impacted by a design like this (but I might be mixing that up with another topic; correct me if I'm wrong).

Reply:
I think the SMTP example was intentionally pushing the envelope on what might be supported/valid, not a known, real use-case.

If the method here is so narrowly scoped, I guess I'll ask: should it be public or private?

If it's only used in this one place, I would tend to rename it to `_handle_eof_errors` or even just inline it here. If it's meant for broader use, and to support subclassing+overriding, then I find it a little odd to have it only used inside of the default implementation of `prompt_for_code`.

Reply:
I'm on board with an underscore prefix. Should I still add a doc string in that case?

Reply:
I just grabbed Derek for a quick sync -- and confirmed that we agree.

The simplest change is to make it private with the underscore. Docstring is optional in that case. This is such a small function, I don't think it's necessary.

---

If we kept it public, we'd probably move it to run outside of `prompt_for_code`. But with it private, the click example I gave becomes:

```python
import contextlib

import click
from globus_sdk.experimental.login_flow_manager import CommandLineLoginFlowManager


class ClickLoginFlowManager(CommandLineLoginFlowManager):
    def print_authorize_url(self, authorize_url: str) -> None:
        click.echo(click.style("Login here for a code:", fg="yellow"))
        click.echo(authorize_url)

    def prompt_for_code(self) -> str:
        try:
            return click.prompt("Enter the code here:")
        except click.Abort as e:
            raise CommandLineLoginFlowEOFError("User aborted when prompted for input.") from e
```

so I think we're on good footing.